vectorizers/preprocessing.py

```diff
@@ -238,11 +244,18 @@ def prune_token_dictionary(
         )
 
     vocab_tokens = [token for token in token_dictionary if token not in tokens_to_prune]
```

Comment:
```suggestion
    vocab_tokens = [token for token in token_dictionary if token not in tokens_to_prune]
    vocab_set = set(vocab_tokens)
```
Sets will provide much faster lookups for "in" requests than lists, so make a set we can query in the next line